src/components/viewmodels/roomlist/RoomListItemViewModel.tsx

```diff
@@ -65,10 +70,14 @@ export function useRoomListItemViewModel(room: Room): RoomListItemViewState {
     const matrixClient = useMatrixClientContext();
     const roomTags = useEventEmitterState(room, RoomEvent.Tags, () => room.tags);
     const isArchived = Boolean(roomTags[DefaultTagID.Archived]);
+    const name = useEventEmitterState(room, RoomEvent.Name, () => room.name);
 
     const notificationState = useMemo(() => RoomNotificationStateStore.instance.getRoomState(room), [room]);
+    // force re-render on notification state change
+    const [, triggerRender] = useState({});
+    useTypedEventEmitter(notificationState, NotificationStateEvents.Update, () => triggerRender({}));
```

Comment:
This is awful, why not just use state for the actual values we care about, in this case looks like a11yLabel & isBold

Reply:
I agree that's awful but in https://github.com/element-hq/element-web/blob/develop/src/components/views/rooms/NotificationDecoration.tsx we rely into a lot of attributes of a `RoomNotificationState`. Extracting all of them is too much imo

Reply:
Sounds like it should use an immutable object which shares the interface it requires, and that object is generated whenever a `NotificationStateEvents.Update` event fires

Reply:
This new immutable object will also trigger a render at every `NotificationStateEvents.Update`. So what is the benefit?

Reply:
The benefit is using the framework we chose as it is designed rather than abusing it in a way which is non-obvious and may break with the upcoming React compiler changes.

Reply:
Why does `NotificationDecoration` not check for `Update` itself? and here then you only need to care about `a11yLabel` & `isBold`

Reply:
https://github.com/element-hq/element-web/blob/develop/src/components/views/rooms/RoomListPanel/RoomListItemView.tsx#L41 we need other value of `notificationState` in the view. I should move their usage in the vm. I think you're right. We need to update 4 values in this vm:

`a11yLabel`, `isBold`, `showHoverMenu` & `isNotificationDecorationVisible`

And `NotificationDecoration`listen to the changes of `RoomNotificationState`